code/game/dna/dna2_helpers.dm

```diff
@@ -133,62 +133,57 @@
+	dna.check_integrity()
+
+	fingerprint        = dna.fingerprint
+	unique_enzymes     = dna.unique_enzymes
+	hair_colour        = rgb(dna.GetUIValueRange(DNA_UI_HAIR_R,255),  dna.GetUIValueRange(DNA_UI_HAIR_G,255),  dna.GetUIValueRange(DNA_UI_HAIR_B,255))
+	facial_hair_colour = rgb(dna.GetUIValueRange(DNA_UI_BEARD_R,255), dna.GetUIValueRange(DNA_UI_BEARD_G,255), dna.GetUIValueRange(DNA_UI_BEARD_B,255))
+	skin_colour        = rgb(dna.GetUIValueRange(DNA_UI_SKIN_R,255),  dna.GetUIValueRange(DNA_UI_SKIN_G,255),  dna.GetUIValueRange(DNA_UI_SKIN_B,255))
+	eye_colour         = rgb(dna.GetUIValueRange(DNA_UI_EYES_R,255),  dna.GetUIValueRange(DNA_UI_EYES_G,255),  dna.GetUIValueRange(DNA_UI_EYES_B,255))
+	skin_tone          = 35 - dna.GetUIValueRange(DNA_UI_SKIN_TONE, 220) // Value can be negative.
+	update_eyes()
+
+	// TODO: update DNA gender to not be a bool - use bodytype and pronouns
+	//Body markings
+	for(var/tag in dna.body_markings)
+		var/obj/item/organ/external/E = GET_EXTERNAL_ORGAN(src, tag)
+		if(E)
+			var/list/marklist = dna.body_markings[tag]
+			if(length(marklist))
+				E.markings = marklist.Copy()
 			else
-				H.set_gender(MALE)
-		*/
-
-		//Body markings
-		for(var/tag in dna.body_markings)
-			var/obj/item/organ/external/E = GET_EXTERNAL_ORGAN(H, tag)
-			if(E)
-				var/list/marklist = dna.body_markings[tag]
-				if(length(marklist))
-					E.markings = marklist.Copy()
-				else
-					LAZYCLEARLIST(E.markings)
-
-		//Base skin and blend
-		for(var/obj/item/organ/organ in H.get_organs())
-			organ.set_dna(H.dna)
-
-		//Hair
-		var/list/hair_subtypes = decls_repository.get_decl_paths_of_subtype(/decl/sprite_accessory/hair)
-		var/hair = dna.GetUIValueRange(DNA_UI_HAIR_STYLE, length(hair_subtypes))
-		if(hair > 0 && hair <= length(hair_subtypes))
-			H.h_style = hair_subtypes[hair]
-
-		//Facial Hair
-		var/list/beard_subtypes = decls_repository.get_decl_paths_of_subtype(/decl/sprite_accessory/facial_hair)
-		var/beard = dna.GetUIValueRange(DNA_UI_BEARD_STYLE, length(beard_subtypes))
-		if((0 < beard) && (beard <= length(beard_subtypes)))
-			H.f_style = beard_subtypes[beard]
-
-		H.force_update_limbs()
-		H.update_body()
-		H.update_eyes()
-		H.update_hair()
-
-		return 1
-	else
-		return 0
+				LAZYCLEARLIST(E.markings)
+
+	//Base skin and blend
+	for(var/obj/item/organ/organ in get_organs())
+		organ.set_dna(dna)
+
+	//Hair
+	var/list/hair_subtypes = decls_repository.get_decl_paths_of_subtype(/decl/sprite_accessory/hair)
+	var/hair = dna.GetUIValueRange(DNA_UI_HAIR_STYLE, length(hair_subtypes))
+	if(hair > 0 && hair <= length(hair_subtypes))
+		h_style = hair_subtypes[hair]
+
+	//Facial Hair
+	var/list/beard_subtypes = decls_repository.get_decl_paths_of_subtype(/decl/sprite_accessory/facial_hair)
+	var/beard = dna.GetUIValueRange(DNA_UI_BEARD_STYLE, length(beard_subtypes))
+	if((0 < beard) && (beard <= length(beard_subtypes)))
+		f_style = beard_subtypes[beard]
+
+	force_update_limbs()
+	update_body()
+	update_eyes()
+	update_hair()
```

Comment:
```suggestion
	update_hair(update_icons = FALSE)
	update_eyes()
```
`update_hair()` should skip the `update_icon()` call because more updates are done after.
`update_eyes()` should go last, because it (for some reason) calls `refresh_visible_overlays()`. (I wouldn't skip `update_hair()` entirely, because it only does so if you have eyes, and not updating hair because you're missing eyes is a bit weird.)